swift/extractor/remapping/SwiftOpenInterception.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+namespace codeql {
+
+void initInterception(const std::string& dir);
+void remapArtifacts(const std::unordered_map<std::string, std::string>& mapping);
```

Comment:
maybe we should consider an alternative API using RAII, something like
```cpp
class Interceptor {
  const std::unordered_map<std::string, std::string>& mapping;

 public:
  // does initInterception, stores reference to mapping
  Interceptor(const std::string& dir, const std::unordered_map<std::string, std::string>& mapping); 

  // does remapArtifacts using stored mapping, does cleanup if needed
  ~Interceptor();
};
```
The mapping does not even need to be filled at construction time.